### PR TITLE
add option: forceNative

### PR DESCRIPTION
**Pull request by finscn**
details at https://github.com/pixijs/pixi.js/issues/4646

### CODE REVIEW

File: src/core/renderers/webgl/WebGLRenderer.js [hunk 1]
```diff
@@ -56,14 +56,16 @@ export default class WebGLRenderer extends SystemRenderer
      *  with older / less advanced devices. If you experiance unexplained flickering try setting this to true.
      * @param {string} [options.powerPreference] - Parameter passed to webgl context, set to "high-performance"
      *  for devices with dual graphics card
+     * @param {Boolean} [options.forceNative=false] - Whether use the native methods or the WebGL extension.
```
Comment (bigtimebuddy) [inline src/core/renderers/webgl/WebGLRenderer.js]: Documentation should also be duplicated in Application and autoDetectRenderer

Comment (themoonrat) [reply src/core/renderers/webgl/WebGLRenderer.js]: and it would be boolean, not Boolean

Comment (finscn) [reply src/core/renderers/webgl/WebGLRenderer.js]: done.

File: src/core/autoDetectRenderer.js [hunk 2]
```diff
@@ -32,6 +32,7 @@ import WebGLRenderer from './renderers/webgl/WebGLRenderer';
  *  If you experience unexplained flickering try setting this to true. **webgl only**
  * @param {string} [options.powerPreference] - Parameter passed to webgl context, set to "high-performance"
  *  for devices with dual graphics card **webgl only**
+ * @param {boolean} [options.forceNative=false] - Whether use the native methods or the WebGL extension.
```
Comment (bigtimebuddy) [inline src/core/autoDetectRenderer.js]: Could you add `**webgl only**` here

File: src/core/Application.js [hunk 3]
```diff
@@ -53,6 +53,7 @@ export default class Application
      *  for devices with dual graphics card **webgl only**
      * @param {boolean} [options.sharedTicker=false] - `true` to use PIXI.ticker.shared, `false` to create new ticker.
      * @param {boolean} [options.sharedLoader=false] - `true` to use PIXI.loaders.shared, `false` to create new Loader.
+     * @param {boolean} [options.forceNative=false] - Whether use the native methods or the WebGL extension.
```
Comment (bigtimebuddy) [inline src/core/Application.js]: Same here, `**webgl only**` comment to this option

Comment (finscn) [reply src/core/Application.js]: done

File: src/core/autoDetectRenderer.js (hunk 2 above)
Comment (finscn) [reply src/core/autoDetectRenderer.js]: done

**Comment by ivanpopelyshev** [comment]
By the way, state is not re-created after context loss. We have to do something about it.

**Comment by bigtimebuddy** [comment]
Why not just do this below? These are all public properties. Would like to not add a convenience setting for a very specific platform bug (old Android).

```js
const app = new PIXI.Application();
app.renderer.state.nativeVaoExtension  = null;
PIXI.glCore.VertexArrayObject.FORCE_NATIVE;
```
I'm in favor of closing this.

**Comment by finscn** [comment]
I don't understand @ivanpopelyshev 's words.
And I agree with "not add a convenience setting for a very specific platform bug"

**Comment by bigtimebuddy** [comment]
Thanks for the PR @finscn. Sorry it took so long to resolve.

**Pull request closed without merging**